src/utils.jl

```diff
@@ -0,0 +1,15 @@
+# an "inside out" kronecker-like product based on broadcasting reshaped arrays
+
+
+# for a single row, some will be scalars, others possibly vectors.  for a whole
+# table, some will be vectors, possibly some matrices
+
+function kron_insideout(op::Function, args...)
+    args = [reshape(a, ones(Int, i-1)..., :) for (i,a) in enumerate(args)]
```

Comment:
You can use a generator rather than a comprehension to avoid an allocation.

I'd also put these functions in the same file as where they are used (since they are called from a single place).